### PR TITLE
OCPQE-12669: Add eus 4.10 to nightly 4.12 upgrade for aws-ipi-ovn

### PR DESCRIPTION
**Pull request by liangxia**
eus 4.10 -> nightly 4.12
versions used, 4.10.37, 4.11.9, 4.12.0-0.nightly-2022-10-25-210451

### CODE REVIEW

File: ci-operator/step-registry/cucushift/upgrade/paused/cucushift-upgrade-paused-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+chain:
+  as: cucushift-upgrade-paused
+  steps:
+    - ref: idp-htpasswd
+    - ref: cucushift-pre
+    - ref: cucushift-upgrade-prepare
+    - ref: cucushift-upgrade-paused-intermediate
+    - ref: cucushift-upgrade-paused-intermediate-sanity
+    - ref: cucushift-upgrade-paused-latest
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/paused/cucushift-upgrade-paused-chain.yaml:9]: For nightly build, this ref is not suitable which use `oc adm upgrade --to-latest` as upgrade command.

Comment (liangxia) [reply ci-operator/step-registry/cucushift/upgrade/paused/cucushift-upgrade-paused-chain.yaml:9]: Would you mind to log an jira issue and show the correct steps in the jira issue ?

Comment (jiajliu) [reply ci-operator/step-registry/cucushift/upgrade/paused/cucushift-upgrade-paused-chain.yaml:9]: sure, I could give the exact eus ci steps in the jira ticket, reviewing what we have currently and what we can re-use here.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    name: "4.10"
+    namespace: ocp
+    tag: upi-installer
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.18
+releases:
+  initial:
```
Comment (shellyyang1989) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:32]: Hmm, seems like the initial release is not used.

File: ci-operator/step-registry/cucushift/upgrade/paused/cucushift-upgrade-paused-chain.yaml (hunk 1 above)
Comment (shellyyang1989) [reply ci-operator/step-registry/cucushift/upgrade/paused/cucushift-upgrade-paused-chain.yaml:9]: Looks like cucushift-upgrade-paused-latest step is using `oc adm upgrade --to-image` [cli](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/paused/latest/cucushift-upgrade-paused-latest-commands.sh#L9) which looks good.

Comment (jiajliu) [reply ci-operator/step-registry/cucushift/upgrade/paused/cucushift-upgrade-paused-chain.yaml:9]: ah, you're right, the `latest` keyword make me confused with `cucushift-upgrade-paused-intermediate-to-latest`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml (hunk 2 above)
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:32]: It's used.
```
time="2022-11-01T01:35:21Z" level=error error="failed to validate configuration /ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml: configuration has 3 errors:
  * tests[0].literal_steps.test[7].dependencies[0]: cannot determine source for dependency \"release:initial\" - this dependency requires a \"initial\" release, which is not configured
  * tests[0].literal_steps.test[9].dependencies[0]: cannot determine source for dependency \"release:initial\" - this dependency requires a \"initial\" release, which is not configured
  * tests[0].literal_steps.test[11].dependencies[0]: cannot determine source for dependency \"release:initial\" - this dependency requires a \"initial\" release, which is not configured"
```

**Comment by liangxia** [comment]
Rebase and fix conflicts.
The tests already finished in earlier test, https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/33558/rehearse-33558-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-eus-4.10-aws-ipi-ovn-p2/1585869958213013504

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml (hunk 2 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: My bad. The latest release is not used should I say. Both initial and latest are the current version of the cluster. We can keep one.

```
yangyang:paused/ (master) $ grep -r RELEASE_IMAGE_LATEST                                                                      [9:50:27]
intermediate/cucushift-upgrade-paused-intermediate-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
latest/cucushift-upgrade-paused-latest-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
unpause/cucushift-upgrade-paused-unpause-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
```

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: It's used, too.
```
time="2022-11-01T01:55:04Z" level=error error="failed to validate configuration /ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml: configuration has 6 errors:
  * tests[0].literal_steps.pre[7].dependencies[0]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.pre[7].dependencies[1]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[1].dependencies[0]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[7].dependencies[2]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[9].dependencies[2]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[11].dependencies[2]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured"
```

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: Hmm, we need to remove the dependencies as well from the yaml files. They are not used in the shell scripts.
intermediate/cucushift-upgrade-paused-intermediate-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
latest/cucushift-upgrade-paused-latest-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
unpause/cucushift-upgrade-paused-unpause-ref.yaml:    env: "RELEASE_IMAGE_LATEST"

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: They are for the backward compatibility.  Some installation use RELEASE_IMAGE_LATEST, while some of the installation use RELEASE_IMAGE_INITIAL, to install the OCP cluster.

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: Good point. I remember we discussed it in the unify CI sync meeting. But I can't remember the decision. Would it be using RELEASE_IMAGE_INITIAL first, if it's not present, using RELEASE_IMAGE_LATEST? @jianlinliu 
Most of our jobs don't specify `initial`. It's kind of weird to have both of them.

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: I did some grep in the whole repo, only `ci-operator/step-registry/ipi/conf/azure/ipi-conf-azure-commands.sh` is using `RELEASE_IMAGE_INITIAL`.
```
function getVersion() {
  local release_image=""
  if [ -n "${RELEASE_IMAGE_INITIAL-}" ]; then
    release_image=${RELEASE_IMAGE_INITIAL}
  elif [ -n "${RELEASE_IMAGE_LATEST-}" ]; then
    release_image=${RELEASE_IMAGE_LATEST}
  fi

  local version=""
  if [ ${release_image} != "" ]; then
    oc registry login
    version=$(oc adm release info ${release_image} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
  fi
  echo "${version}"
}
```
But it also support `RELEASE_IMAGE_LATEST`, it looks like it also works without `RELEASE_IMAGE_INITIAL`.  I am not sure if there would be some other backward compatibility issues.

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: Anyway, it should not be a blocker for this PR. We can use a follow up PR to try to remove that (and related references in the steps). WDYT ?

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml:40]: Sure.

**Pull request merged**